applications/OptimizationApplication/python_scripts/execution_policies/kratos_analysis_execution_policy.py

```diff
@@ -0,0 +1,114 @@
+        self.parameters.ValidateAndAssignDefaults(default_settings)
+
+        analysis_module = parameters["analysis_module"].GetString()
+        analysis_type = parameters["analysis_type"].GetString()
+        analysis_settings = parameters["analysis_settings"]
+
+        if analysis_module == "KratosMultiphysics":
+            analysis_module = GetClassModuleFromKratos(analysis_type)
+
+        self.model_parts: 'list[Kratos.ModelPart]' = []
+        analysis_full_module = f"{analysis_module}.{Kratos.StringUtilities.ConvertCamelCaseToSnakeCase(analysis_type)}"
+        self.analysis: AnalysisStage = getattr(import_module(analysis_full_module), analysis_type)(self.model, analysis_settings.Clone())
+
+        self.nodal_solution_step_data_variables = []
+        if self.parameters["analysis_output_settings"].Has("nodal_solution_step_data_variables"):
+            self.nodal_solution_step_data_variables = self.parameters["analysis_output_settings"]["nodal_solution_step_data_variables"]
+        self.nodal_data_value_variables = []
+        if self.parameters["analysis_output_settings"].Has("nodal_data_value_variables"):
+            self.nodal_data_value_variables = self.parameters["analysis_output_settings"]["nodal_data_value_variables"]
+        self.element_data_value_variables = []
+        if self.parameters["analysis_output_settings"].Has("element_data_value_variables"):
+            self.element_data_value_variables = self.parameters["analysis_output_settings"]["element_data_value_variables"]
+        self.condition_data_value_variables = []
+        if self.parameters["analysis_output_settings"].Has("condition_data_value_variables"):
+            self.condition_data_value_variables = self.parameters["analysis_output_settings"]["condition_data_value_variables"]
+
+
+    def GetAnalysisModelPart(self):
+        return self.analysis._GetSolver().GetComputingModelPart()
+
+    def Initialize(self):
+        self.analysis.Initialize()
+
+        # initialize model parts
+        self.model_parts = [self.model[model_part_name] for model_part_name in self.parameters["model_part_names"].GetStringArray()]
+
+    def Check(self) -> None:
+        pass
+
+    def Finalize(self) -> None:
+        self.analysis.Finalize()
+
+    def Execute(self):
+        self.analysis.time = self.analysis.project_parameters["problem_data"]["start_time"].GetDouble()
+        for model_part in self.model_parts:
+            model_part.ProcessInfo.SetValue(Kratos.STEP, 0)
+            model_part.ProcessInfo.SetValue(Kratos.TIME, 0)
+            model_part.ProcessInfo.SetValue(Kratos.DELTA_TIME, 0)
+        self.analysis.RunSolutionLoop()
+
+        optimization_problem_data = self.optimization_problem.GetProblemDataContainer()
+        if optimization_problem_data.HasValue("requested_container_expression_outputs") and self in optimization_problem_data["requested_container_expression_outputs"]:
+            self._OutputAnalysisData()
```

Comment:
We don't need this check anymore ryt? we can check `self.nodal_solution_step_data_variables`, etc variables, and write accordingly which is done in `_OutputAnalysisData`. So If the user gives an empty list, this does not do anything.